core/vm/context/context.go

```diff
@@ -0,0 +1,142 @@
+package context
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/consensus/istanbul"
+	"github.com/celo-org/celo-blockchain/contracts"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/log"
+)
+
+// New creates a new context for use in the EVM.
+func New(msg vm.Message, header *types.Header, chain vm.ChainContext, txFeeRecipient *common.Address) vm.Context {
```

Comment:
I like that you removed `TobinTransfer` and the evm callers from `evm.go`, but I think moving all of the context will make upstream harder to merge.

Reply:
We could re-instate inside evm.go the NewEVMContext from upstream and then additionally have this version here which we will use. That would help push the vm package closer to geth

Reply:
I though on this. To me It's a balance. By removing almost all our changes to `evm.go` i made much closer to upstream and simpler to merge. Now our changes are centralized in the `context.go`. I'd argue that's better since you isolated your changed to a single place. And it's not a very large file, so i don't see it very challenging.

I also though of leaving the old NewEVMContext and creating a new on top, but i don't like that, because the old version will just create an invalid version of a vm.Context and i'd like the construct to validate the context is actually correct. I might be encouraging human error if you use the old version by mistake. So i prefer to remove that version alltogheter.

wdyt?

Reply:
I agree that I don't like having two `NewEVMContext` functions. Even if there are conflicts in merging upstream in the `evm.go:NewEVMContext`, it still feels easier than manually porting over the changes to the new file.

Reply:
Agreed

Reply:
bad news....:P `vm/context.go` doesn't really exists on upstream :P 

It seems we've already moved it from `core/evm.go` to `core/vm/context.go`! With that in mind, i think i'll just move all the remainig code within vm/context.go to the context pkg